mux.go

```diff
@@ -152,6 +157,23 @@ func (r *Router) SkipClean(value bool) *Router {
 	return r
 }
 
+// UseEncodedPath defines the route matching behavior. The initial value is
+// false.
+//
+// When true, the router will match the encoded original path to the routes.
+// For eg. "/path/foo%2Fbar/to" will match the path "/path/{var}/to".
+// This behavior has the drawback of needing to match routes against
+// r.RequestURI instead of r.URL.Path. Any modifications (such as http.StripPrefix)
+// to r.URL.Path will not affect routing when this flag is on and thus may
+// induce unintended behavior.
+//
+// When false, the router will match the unencoded path to the routes.
+// For eg. "/path/foo%2Fbar/to" will match the path "/path/foo/bar/to"
+func (r *Router) UseEncodedPath(value bool) *Router {
```

Comment:
Silly edge case but what if someone wants to turn it off later? They turn it on, add a couple of routes, turn it off and add some routes they don't want the behavior for.